test/integration/interrupt_resume_test.rb

```diff
@@ -304,4 +303,61 @@ def test_interrupt_resume_inline_verifier_will_verify_additional_rows_changed_on
     error_line = ghostferry.error_lines.last
     assert_equal "cutover verification failed for: gftest.test_table_1 [paginationKeys: #{chosen_id} ] ", error_line["msg"]
   end
+
+  # originally taken from @kolbitsch-lastline in https://github.com/Shopify/ghostferry/pull/160
+  def test_interrupt_resume_between_consecutive_rows_events
+    ghostferry = new_ghostferry(MINIMAL_GHOSTFERRY, config: { verifier_type: "Inline" })
+
+    start_binlog_status = source_db.query('SHOW MASTER STATUS').first
+
+    # create a series of rows-events that do not have interleaved table-map
+    # events. This is the case when multiple rows are affected in a single
+    # DML event.
+    # Since we are racing between applying rows and sending the shutdown event,
+    # we emit a whole bunch of them
+    num_batches = 20
+    num_values_per_batch = 1000
+    row_id = 0
+    ghostferry.on_status(Ghostferry::Status::BINLOG_STREAMING_STARTED) do
+      for _batch_id in 0..num_batches do
+        insert_sql = "INSERT INTO #{DEFAULT_FULL_TABLE_NAME} (data) VALUES "
+        for value_in_batch in 0..num_values_per_batch do
+          row_id += 1
+          insert_sql += ", " if value_in_batch > 0
+          insert_sql += "('data#{row_id}')"
+        end
+        source_db.query(insert_sql)
+      end
+    end
+
+    ghostferry.on_status(Ghostferry::Status::AFTER_BINLOG_APPLY) do
+      # while we are emitting events in the loop above, try to inject a shutdown
+      # signal, hoping to interrupt between applying an INSERT and receiving the
```

Comment:
So this test will be known to racy? Will it make CI more flaky?

Reply:
It has been consistently passing in my experience and hasn’t been flaky